gateway/gateway.go

```diff
@@ -674,18 +687,39 @@ func (gateway *HandleT) getJobFromRequest(req *webRequestT) (job *jobsdb.JobT, n
 		)
 	}
 	err = nil
-	job = &jobsdb.JobT{
+	job := &jobsdb.JobT{
 		UUID:         id,
 		UserID:       builtUserID,
 		Parameters:   marshalledParams,
 		CustomVal:    CustomVal,
 		EventPayload: body,
-		EventCount:   numEvents,
+		EventCount:   jobData.numEvents,
 		WorkspaceId:  workspaceId,
 	}
+	jobData.job = job
 	return
 }
 
+func (*HandleT) getSourceDefForWriteKey(writeKey string) string {
+	configSubscriberLock.RLock()
+	defer configSubscriberLock.RUnlock()
+
+	if _, ok := writeKeysSourceMap[writeKey]; ok {
+		return writeKeysSourceMap[writeKey].SourceDefinition.Category
```

Comment:
what does a `null` category refer to? Shouldn't we return a value instead of an empty string for that case?

Reply:
returning `eventStream` in case of null categories.

Reply:
if `writeKey` not found in configMap, returns an empty string.